krate.go

```diff
@@ -0,0 +1,297 @@
+	device_address string
+	path           string
+	cmd            string
+}
+
+const tftSetup string = `
+#define USER_SETUP_INFO "User_Setup"
+#define ST7789_DRIVER
+#define TFT_RGB_ORDER TFT_BGR
+#define TFT_WIDTH  240
+#define TFT_HEIGHT 240
+#define TFT_MISO -1
+#define TFT_MOSI 23
+#define TFT_SCLK 18
+#define TFT_CS 5
+#define TFT_DC 2
+#define TFT_RST 4
+#define TFT_BL 15
+#define LOAD_GLCD
+#define LOAD_FONT2
+#define LOAD_FONT4
+#define LOAD_FONT6
+#define LOAD_FONT7
+#define LOAD_FONT8
+#define LOAD_GFXFF
+#define SMOOTH_FONT
+#define SPI_FREQUENCY 1000000
+#define SPI_READ_FREQUENCY 27000000
+#define SPI_TOUCH_FREQUENCY 2500000
+`
+
+const mainCpp string = `
+#include <Arduino.h>
+#include <otaserver.h>
+#include <kgfx.h>
+
+OTAServer otaserver;
+KGFX ui;
+
+void setup() {
+  Serial.begin(460800);
+  Serial.println("Starting app");
+
+  otaserver.connectWiFi(); // DO NOT EDIT.
+  otaserver.run(); // DO NOT EDIT
+
+  ui.init();
+  ui.clear();
+  ui.drawText("hello", Arial_28, TFT_YELLOW, 0, 0);
+  Serial.println("Drew hello");
+}
+
+void loop() {
+  if((WiFi.status() == WL_CONNECTED)) {
+    otaserver.handle(); // DO NOT EDIT
+  }
+
+  delay(1);
+}
+`
+
+const pioini string = `
+; PlatformIO Project Configuration File
+;
+;   Build options: build flags, source filter
+;   Upload options: custom upload port, speed and extra flags
+;   Library options: dependencies, extra library storages
+;   Advanced options: extra scripting
+;
+; Please visit documentation for the other options and examples
+; https://docs.platformio.org/page/projectconf.html
+
+[env:esp32dev]
+platform = espressif32
+board = esp32dev
+framework = arduino
+lib_deps =
+	bodmer/TFT_eSPI@^2.5.0
+	kublet/KGFX@^0.0.14
+	kublet/OTAServer@^1.0.4
+monitor_speed = 460800
+`
+
+/**
+ * Initialize the Krate struct
+ */
+func (k *Krate) Initialize() error {
+	user, err := user.Current()
+	if err != nil {
+		return err
+	}
+	k.username = user.Username     // get the current user
+	if runtime.GOOS != "windows" { // Non-Windows Path
+		k.path = filepath.Join(user.HomeDir, ".platformio", "penv", "bin")
+	} else { // Windows Path
+		k.path = filepath.Join(user.HomeDir, ".platformio", "penv", "Scripts")
+	}
+	k.cmd = "pio" // the command to run
+	k.device_address = ""
+	return nil
+}
+
+/**
+* print help message
+ */
+func (k *Krate) Help() {
+	fmt.Println("Usage: krate [options...] <arg>")
+	fmt.Println(" help                    List all available commands")
+	fmt.Println(" init                    Initializes project with basic libraries and code")
+	fmt.Println(" deps install            Install dependencies specified in platformio.ini file")
+	fmt.Println(" build                   Compiles project")
+	fmt.Println(" send <ip address>       Send firmware OTA to kublet by specifiying ip address")
+	fmt.Println(" monitor                 Monitors logs")
+	fmt.Println(" publish									Creates manifest file based on prompts and user inputs for publishing on Kublet community")
```

Comment:
Not sure why this is still showing up as empty space. Let's merge it first and I'll fix it.

Reply:
Probably a GitHub issue, resolved.